video/repack.c

```diff
@@ -157,8 +157,8 @@ static void swap_endian(struct mp_image *dst, int dst_x, int dst_y,
         assert(src->fmt.bpp[p] == bpp * 8);
 
         for (int y = 0; y < h; y++) {
-            void *s = mp_image_pixel_ptr_ny(src, p, src_x, src_y + y);
-            void *d = mp_image_pixel_ptr_ny(dst, p, dst_x, dst_y + y);
+            void *restrict s = mp_image_pixel_ptr_ny(src, p, src_x, src_y + y);
+            void *restrict d = mp_image_pixel_ptr_ny(dst, p, dst_x, dst_y + y);
```

Comment:
thought: if we turn these three into static inline functions in the header `restrict` should automatically propagate to derived pointers
```c
void *mp_image_pixel_ptr(struct mp_image *img, int plane, int x, int y);
void *mp_image_pixel_ptr_ny(struct mp_image *img, int plane, int x, int y);
size_t mp_image_plane_bytes(struct mp_image *img, int plane, int x0, int w);
```

Reply:
I don't think it would propagate, even if we mark `mp_image` itself restrict, the derived pointers are something else and cannot assume they are distinct.

Reply:
yeah I have no idea where I was going with this suggestion.